unpacker.go

```diff
@@ -187,16 +187,18 @@ func (u *unpacker) unpack(ctx context.Context, config ocispec.Descriptor, layers
 }
 
 func (u *unpacker) handlerWrapper(uctx context.Context, unpacks *int32) (func(images.Handler) images.Handler, *errgroup.Group) {
```

Comment:
Instead of returning the whole error group, maybe return a function which will wait then cancel?

Reply:
eg itself does wait and cancel.

We need cancel and wait when fetch returns error.

Reply:
Sorry, I should have been more clear. I agree you need to explicitly cancel on error. However cancel must always be run, and if it hasn't been canceled, then do so after wait (or just always do after wait since double cancellation is fine)